src/jit/codegenxarch.cpp

```diff
@@ -5579,6 +5579,13 @@ void CodeGen::genCallInstruction(GenTreeCall* call)
         instGen(INS_vzeroupper);
     }
 
+    if (callType == CT_HELPER && compiler->info.compFlags & CORINFO_FLG_SYNCH)
+    {
+        fPossibleSyncHelperCall = true;
```

Comment:
Is the code that sets `fPossibleSyncHelperCall` below redundant with this change?

Reply:
It would appear so to me; the code below should probably be removed.